Pharo 9 + Seaside - Process dying on cloud instance

A few weeks back it was discussed on Discord and Pharo ML what would be minimal costs to run a Pharo web application online.
So I wanted to check on Azure cloud from Microsoft with a minimal Linux VM running what would be the average costs in running something with Pharo using the cloud.

To perform this test I used Pharo and Seaside on a cloud virtual machine (using Microsoft Azure). The Seaside image is exposed on a public IP address. I've let it run overnight and every second day the Pharo process/virtual machine was gone. It died inbetween which was reproducible. Unfortunately Pharo left no traces why it died - there was no debug log or other.

Maybe it is related to the resource restricted tiny Linux VM with not much RAM. 

Peter Svenson tried the same with similar results.

Hmm, my VM (from get.pharo.org/90+vm) gives:
```
sven@stfx-1:~/pharo$ ./pharo --mmap 128m Pharo.image eval 42 factorial
[ERROR] 2021-07-13 13:03:53.000 processVMOptions (src/parameters.c:459):Invalid or unknown VM parameter --mmap
Usage: Pharo [<option>...] [<imageName> [<argument>...]]
       Pharo [<option>...] -- [<argument>...]

Common <option>s:
  --help                 	Print this help message, then exit
  --headless             	Run in headless (no window) mode (default: false)
  --worker               	run in worker thread (default: false)
  --logLevel=<level>     	Sets the log level (ERROR, WARN, INFO or DEBUG)
  --version              	Print version information, then exit
  --maxFramesToLog=<cant>	Sets the max numbers of Smalltalk frames to log
Notes:

  <imageName> defaults to `Pharo.image'.
  <argument>s are ignored, but are processed by the Pharo image.
  Precede <arguments> by `--' to use default image.
```
This strikes me as a very low number of options ...

It already has some trouble printing the version:

```
myuser@TESTPHARO-LINUX-VM-001:~$ mkdir seaside
myuser@TESTPHARO-LINUX-VM-001:~$ cd seaside/
myuser@TESTPHARO-LINUX-VM-001:~/seaside$ curl get.pharo.org/90+vm | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3036  100  3036    0     0  52519      0 --:--:-- --:--:-- --:--:-- 53263
Downloading the latest 90 Image:
    http://files.pharo.org/get-files/90/pharoImage-x86_64.zip
Pharo.image
Downloading the latest pharoVM:
        http://files.pharo.org/get-files/90/pharo-vm-Linux-x86_64-stable.zip
pharo-vm/pharo
Creating starter scripts pharo and pharo-ui
myuser@TESTPHARO-LINUX-VM-001:~/seaside$ rm pharo-ui
myuser@TESTPHARO-LINUX-VM-001:~/seaside$ ./pharo Pharo.image printVersion
[ERROR] 2021-07-19 11:17:28.000 sqAllocateMemory (src/memoryUnix.c:157):I cannot find a good memory address starting from: 0x7f0e182a5000
```

--- a/src/UIManager/CommandLineUIManager.class.st
+++ b/src/UIManager/CommandLineUIManager.class.st
@@ -188,8 +188,13 @@ CommandLineUIManager >> exitFailure [
 
 { #category : #'handle debug requests' }
 CommandLineUIManager >> handleDebugRequest: aDebugRequest [
-	
-	self unhandledErrorDefaultAction: aDebugRequest exception
+
+	self
+		debugProcess: aDebugRequest process
+		context: aDebugRequest exception signalerContext
+		label: aDebugRequest exception description
+		fullView: false
+		notification: aDebugRequest exception description
 ]
 
 { #category : #'error handling' }

